### PR TITLE
Add host and port to BrokerPool cache key

### PR DESCRIPTION
**Pull request by djiang**
### Problem
In cases where a Kafka broker goes down and new broker with a different host/port spins up with the same broker id, we don't want to use the previously cached broker because the connection is still pointing to the old host/port.

### Solution
To fix this, we'll consider the host and port when deciding whether or not to use the cached broker. If the cached broker is no longer valid, we'll disconnect and close the connection.

### CODE REVIEW

**Comment by dasch** [comment]
:heart:

Could you add a unit test for this?

**Comment by djiang** [comment]
Hey @dasch, I added some `BrokerPool` specs 😃

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.2.3
```
Comment (dasch) [inline .ruby-version]: Can you remove this file from the PR? It doesn't seem related.

**Comment by dasch** [comment]
It would also be great if you could give this branch a test in a production environment for a while before I merge, as I'm unable to do so myself for the time being. This change is rather risky.

**Comment by dasch** [comment]
@osheroff could you also take a look? I won't have much internet the next few days.

File: .ruby-version (hunk 1 above)
Comment (djiang) [inline .ruby-version]: Sure thing

**Comment by djiang** [comment]
I'll keep my branch running for my project over the weekend.

**Comment by djiang** [comment]
I think I found some issues -- I'll investigate and report back

**Comment by dasch** [comment]
Thanks!

**Comment by djiang** [comment]
The issue seems to lie in the cluster not refreshing the metadata when it gets into certain states. The fix I have in place works, but it's a bit defensive. I want to spend some more time to get to the root cause of the problem.

**Comment by dasch** [comment]
@djiang could that be related to #294?

**Comment by djiang** [comment]
Hmm I don't think #294 is related as it pertains to the producer and the issue here is in the consumer.

I'll dig more into this issue soon -- I haven't had a lot of time recently.

**Comment by dasch** [comment]
@djiang do you have an update on this?

**Comment by dasch** [comment]
@djiang any update, or should I close this issue?

**Comment by djiang** [comment]
We can close this for now -- haven't had a chance to revisit.

**Comment by dasch** [comment]
👍
